docs/source/step-by-step.rst

```diff
@@ -142,14 +143,16 @@ this:
         5    |     0  |     0  |   n/a   |            0
         6    |     0  |     0  |   n/a   |            0
         7    |     0  |     0  |   n/a   |            0
-       183M (   183087673) cycles total
+       146M (   145699910) cycles total
```

Comment:
I'm not able to replicate this decrease in cycle count with my setup (approx. 20% decrease seems significant), do you know why this could be different for you and I? If I can replicate this, I'll make the numbers consistent throughout the doc.

Reply:
Yes I wondered about that. I thought I should make the example output consistent with itself (I thought it would be bad to show a CSV from my faster run with the total cycles from a slower run that wouldn't add up).

Have there been some other recent improvements on the main branch which could have sped things up? Submodule bump in Vexriscv perhaps?

I am using Vivado 2020.2, although I don't see how any timing improvement in PnR could result in fewer cycles here.

Could my Arty board have faster DDR or something like that? Seems implausible. This board says it's rev E.

Reply:
I also have rev E. I'll go ahead and test again directly from main (with a different Arty as well). I vaguely remember you mentioning something about building a more recent toolchain, doubt that would cause such a dramatic change though. Worst case scenario, I can update the numbers with my run so that they can be internally consistent and keep your explanation.

Reply:
Hmm I have gcc 10.2, not 8.x as the instructions in the docs recommend.

Reply:
I double-checked the timing, built `proj_template` from main branch commit e9f6c26cf8151a4cd568c569abfef4ae15b207d2, no local modifications:

```
   146M (   145807043) cycles total
```

Reply:
Ah ha. I tried it again using gcc 8.3 from the prebuilt toolchain mentioned in the docs.

```
   183M (   183328857) cycles total
```

I'm surprised the difference is so drastic. Maybe this means we should update to point at a newer toolchain. I see that Sifive have published prebuilt binaries for gcc 10.2 now.

Reply:
Wow, that performance difference is no joke. Definitely something to move near the top of the to-do list, thanks so much for figuring this out :).